### PR TITLE
go.mod: patch with lib/pq TimeTZ fix 

### LINKED ISSUE #65066: cli: timetz with second offset does not display on CLI
timetz with second offsets behave poorly on the CRDB CLI:

```
demo@127.0.0.1:26257/defaultdb> select * from ( VALUES ('12:00'::timetz), ( '12:05:00+01:01:05'::timetz)  ) tbl(t);
          t
---------------------
  12:00:00+00:00:00
(1 row)
(error encountered after some results were delivered)
ERROR: pq: decode: parsing time "12:05:00+01:01:05": extra text: ":05"
demo@127.0.0.1:26257/defaultdb> select * from ( VALUES ( '12:05:00+01:01:05'::timetz), ('12:00'::timetz) ) tbl(t);
  t
-----
(0 rows)

Time: 1ms
warning: pq: unexpected DataRow in simple query execution
ERROR: driver: bad connection
warning: connection lost!
opening new connection: all session settings will be lost
```

psql on CRDB:

```
root=# select * from ( VALUES ('12:00'::timetz), ( '12:05:00+01:01:05'::timetz)  ) tbl(t);
         t
-------------------
 12:00:00+00:00
 12:05:00+01:01:05
(2 rows)
```

### PR DESCRIPTION
**Pull request by otan**
Resolves #65066.

See individual commits for details.

### CODE REVIEW

File: DEPS.bzl [hunk 1]
```diff
@@ -2429,8 +2429,9 @@ def go_deps():
         name = "com_github_lib_pq",
         build_file_proto_mode = "disable_global",
         importpath = "github.com/lib/pq",
-        sum = "h1:9xohqzkUwzR4Ga4ivdTcawVS89YSDVxXMa3xJX3cGzg=",
-        version = "v1.8.0",
+        replace = "github.com/cockroachdb/pq",
```
Comment (rafiss) [inline DEPS.bzl:2432]: https://github.com/lib/pq/pull/1039 was merged so should we just tag/release that?

File: pkg/sql/logictest/testdata/logic_test/timetz [hunk 2]
```diff
@@ -670,3 +670,8 @@ SELECT a FROM regression_44774 ORDER BY a
 
 statement ok
 DROP TABLE regression_44774
+
+query T
+SELECT '20:01:02+03:04:05' AS regression_65066
```
Comment (rafiss) [inline pkg/sql/logictest/testdata/logic_test/timetz:675]: does this test catch it? i thought the issue was in the CLI so we'd need something in cli_test.go?

(when i connect to crdb using psql, this query works fine on master. do logic tests use the cli formatting?)

**Review by rafiss — CHANGES_REQUESTED** [review]

File: DEPS.bzl (hunk 1 above)
Comment (otan) [reply DEPS.bzl:2432]: see the comment in go.mod below -- we can't use lib/pq on master because of #65320 ...

File: pkg/cli/cli_test.go [hunk 3]
```diff
@@ -196,6 +196,7 @@ func Example_sql() {
 	// and with the query below the division by zero error will occur after the
 	// first batch consisting of 1 row has been returned to the client.
 	c.RunWithArgs([]string{`sql`, `-e`, `select 1/(@1-2) from generate_series(1,3)`})
+	c.RunWithArgs([]string{`sql`, `-e`, `SELECT '20:01:02+03:04:05' AS regression_65066`})
```
Comment (rafiss) [inline pkg/cli/cli_test.go:199]: oops sorry i missed before, but we'd need the `::timetz` cast here right? otherwise it is just a string value.

**Review by rafiss — CHANGES_REQUESTED** [review]
one more issue, then lgtm -- though as i mentioned i'll poke around at the other issue that's blocking the lib/pq upgrade first

File: pkg/cli/cli_test.go (hunk 3 above)
Comment (otan) [reply pkg/cli/cli_test.go:199]: my bad

**Review by rafiss — APPROVED** [review]
lgtm! let's merge and i hope i can come back and look at https://github.com/cockroachdb/cockroach/issues/65320 later

> Reviewed 3 of 4 files at r1, 1 of 2 files at r3, 1 of 1 files at r5.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/65321#-:-MadwKxb-O5Bhu5BPeEC:bhnts1x)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan and @rafiss)

**Comment by otan** [comment]
bors r=rafiss

**Comment by otan** [comment]
bors r=rafiss

**Comment by otan** [comment]
bors r=rafiss

**Pull request merged**